devito/interfaces.py

```diff
@@ -151,7 +152,10 @@ def _indices(cls, **kwargs):
                 raise ValueError("Unknown symbol dimensions or shape")
             _indices = [x, y, z]
             shape = kwargs.get('shape')
-            dimensions = _indices[:len(shape)]
+            if len(shape) <= 3:
+                dimensions = _indices[:len(shape)]
+            else:
+                dimensions = [symbols("x%d" % i) for i in range(1, len(shape) + 1)]
```

Comment:
Or we could simply *fix the broken test* by using a `TimeData` object that auto-inserts the `t` dimension that for some reason is forced into the `DenseData` constructor. To be clear though, I'm not opposed to making the Propagator more generic (although I believe there is more to it that the change we're discussing here), but there is nothing in our test base that warrants this to happen as part of a completely unrelated feature - except for the test you pointed out, which is clearly using the symbolic objects in an unintended way.